Makefile

```diff
@@ -16,87 +12,147 @@ endif
 	docker-compose down -v
 	@echo "${RED}[#] The local Docker project was removed${RESET}"
 
+# RUN ONLY ON FRESH VM THAT HAS NOTHING (amd64 Debian buster)
+# [1] Runs all the commands here: https://docs.docker.com/engine/install/debian/
+#     to install Docker
+# [2] Runs all the commands here: https://docs.docker.com/compose/install/
+#     to install docker-compose
+# [3] Logs in to gcloud
+# [4] Install sops
+d_prod_install_all:
+	@echo "${GREEN}[# 1] Installing Docker... ${RESET}"
+	sudo apt-get update
+	sudo apt-get install \
+		apt-transport-https \
+		ca-certificates \
+		curl \
+		gnupg \
+		lsb-release -y
+	curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+	echo \
+		"deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian \
+		buster stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+	sudo apt-get update
+	sudo apt-get install docker-ce docker-ce-cli containerd.io -y
+	@echo "${GREEN}[# 1] Running Docker hello world ${RESET}"
+	sudo docker run hello-world
+	@echo "${GREEN}[# 1] If you saw the hello world, you are done :) ${RESET}"
+
+	@echo "${GREEN}[# 2] Installing docker-compose ...${RESET}"
+	sudo curl -L "https://github.com/docker/compose/releases/download/1.29.1/docker-compose-Linux-x86_64" -o /usr/local/bin/docker-compose
+	sudo chmod +x /usr/local/bin/docker-compose
+	docker-compose --version
+	@echo "${GREEN}[# 2] If you saw the docker-compose version, you are done :) ${RESET}"
+
+	@echo "${GREEN}[# 3] Loging in to gcloud please follow terminal instructions ${RESET}"
+	gcloud auth login
+	gcloud auth application-default login
+	gcloud auth configure-docker
+	gcloud auth list
+	@echo "${GREEN}[# 3] If you see yourself on the ACTIVE ACCOUNT your are done :) ${RESET}"
+
+	@echo "${GREEN} [# 4] Installing sops ... ${RESET}"
+	curl -j -O -L https://github.com/mozilla/sops/releases/download/v3.7.1/sops_3.7.1_amd64.deb
+	apt install ./sops_3.7.1_amd64.deb
+	rm ./sops_3.7.1_amd64.deb
+	@echo "${GREEN} [# 4] Sops installed ${RESET}"
+
+	@echo "If all is green and worked run 'make decrypt_env && make d_prod_install'"
+
 d_prod_install:
 	@echo "${GREEN}[#] STARTING RPOD DOCKER INSTALL${RESET}"
 	docker pull nginx:1.19.0-alpine
+	export CLOUDSDK_PYTHON=python3
 	chmod +x entrypoint.prod.sh
-	docker-compose -f docker-compose.prod.yml build
-	docker-compose -f docker-compose.prod.yml up -d --build
-	docker-compose exec web python manage.py migrate --noinput
-	docker-compose exec web python manage.py createsuperuser --noinput
-	docker-compose exec web python manage.py collectstatic --noinput
+	docker pull eu.gcr.io/victor-ciurana-com/victorciuranacom_web:latest
+	docker pull eu.gcr.io/victor-ciurana-com/victorciuranacom_nginx-proxy:latest
+	# docker-compose -f docker-compose.prod.yml pull
+	docker-compose -f docker-compose.prod.yml up -d --force-recreate --remove-orphans
```

Comment:
Okay, I think here is where you miss the point of deploying with docker. You are not supposed to run these commands in your VM :). So you don't have to ssh to it, git pull the project there and then run the docker commands.

The only thing you need to do is:

`sops exec-env .prod.env 'docker-compose -f docker-compose.prod.yml -H ssh://user@host -d --force-recreate --remove-orphans'`

This will, load the environment variables you have in the encrypted sops file and pass them to the `docker-compose` command. The `docker-compose` command will use `ssh` (with whatever private key you have set up) to connect remotely to the `docker` daemon you have running in your VM and deploy the services you have set up in the `docker-compose.prod.yml` file.

Reply:
Makes sense, I'll be trying that oneliner.